### PR TITLE
Support multiple datasources for reactive SQL clients

### PR DESCRIPTION
**Pull request by gsmet**
In passing, also removed the long time deprecated Axle and Rx wrappers (I checked with @cescoffier first).

/cc @Sanne @cescoffier @tsegismont

### CODE REVIEW

**Comment by gsmet** [comment]
Still draft while running a full CI run in my fork.

**Comment by gsmet** [comment]
CI all green in my fork, this is ready for review!

File: extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/DataSourcesReactiveDB2Config.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * For now, the reactive extensions only support a default datasource.
+ */
+@ConfigRoot(name = "datasource", phase = ConfigPhase.RUN_TIME)
+public class DataSourcesReactiveDB2Config {
+
+    /**
+     * The default datasource.
+     */
+    @ConfigItem(name = "reactive.mysql")
```
Comment (machi1990) [inline extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/DataSourcesReactiveDB2Config.java:22]:
```suggestion
    @ConfigItem(name = "reactive.db2")
```

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourcesReactiveRuntimeConfig.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import io.quarkus.runtime.annotations.ConfigDocMapKey;
+import io.quarkus.runtime.annotations.ConfigDocSection;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * For now, the reactive extensions only support a default datasource.
+ */
```
Comment (machi1990) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourcesReactiveRuntimeConfig.java:15]: This is no longer true. I noticed it throughout.

File: extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/health/ReactiveDB2DataSourcesHealthCheck.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.datasource.common.runtime.DataSourceUtil;
+import io.quarkus.reactive.datasource.ReactiveDataSource;
+import io.vertx.mutiny.db2client.DB2Pool;
+
+@Readiness
+@ApplicationScoped
+public class ReactiveDB2DataSourcesHealthCheck implements HealthCheck {
```
Comment (machi1990) [inline extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/health/ReactiveDB2DataSourcesHealthCheck.java:26]:
```suggestion 
class ReactiveDB2DataSourcesHealthCheck implements HealthCheck {
```

File: extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/health/ReactiveMySQLDataSourcesHealthCheck.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.datasource.common.runtime.DataSourceUtil;
+import io.quarkus.reactive.datasource.ReactiveDataSource;
+import io.vertx.mysqlclient.MySQLPool;
+
+@Readiness
+@ApplicationScoped
+public class ReactiveMySQLDataSourcesHealthCheck implements HealthCheck {
```
Comment (machi1990) [inline extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/health/ReactiveMySQLDataSourcesHealthCheck.java:27]:
```suggestion 
class ReactiveMySQLDataSourcesHealthCheck implements HealthCheck {
```

File: extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/health/ReactivePgDataSourcesHealthCheck.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.datasource.common.runtime.DataSourceUtil;
+import io.quarkus.reactive.datasource.ReactiveDataSource;
+import io.vertx.pgclient.PgPool;
+
+@Readiness
+@ApplicationScoped
+public class ReactivePgDataSourcesHealthCheck implements HealthCheck {
```
Comment (machi1990) [inline extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/health/ReactivePgDataSourcesHealthCheck.java:27]:
```suggestion 
class ReactivePgDataSourcesHealthCheck implements HealthCheck {
```

**Review by machi1990 — DISMISSED** [review]
Mostly nitpicks and one feedback about changing configuration knob for DB2 database from `reactive.myql` to `reactive.db2`.

**Comment by gsmet** [comment]
@machi1990 thanks for checking. I fixed a few other minor details based on your comments and added a paragraph in the doc in a second commit.

**Comment by tsegismont** [comment]
@gsmet thanks for pinging adding me in cc. I will review the PR this morning.

**Review by tsegismont — APPROVED** [review]
LGTM, thank you for taking care of this @gsmet !

**Comment by Sanne** [comment]
That's great! thanks all

**Pull request merged**